apps/emqx/src/emqx_topic.erl

```diff
@@ -33,19 +35,6 @@
     parse/2
 ]).
 
--export_type([
```

Comment:
we should try to avoid using header files for type specs, but not the other way around.
using headers increases the chance of duplication specs.

Reply:
Specification conflicts will be reported and resolved at compile time.

In my opinion moving the type spec into a single header file avoids duplication of specification defins as much as possible. Modern LSP can also find all references to a type definition, avoiding cases where a code change misses it. 🤔

In this case, using type definitions in header files has no effect on using these types in imported functions.